packages/react-router-dom/dom.ts

```diff
@@ -258,12 +283,19 @@ export function getFormSubmissionInfo(
           formData.append(name, value);
         }
       } else if (target != null) {
+        // When a raw object is sent - even though we encode it into formData,
+        // we still expose it as payload so it aligns with the behavior if
+        // encType were application/json or text/plain
+        payload = target;
```

Comment:
@ryanflorence I went slightly against the mutual exclusivity here so that `application/x-www-form-urlencoded` would better mimic the behavior of `application/json` and `text/plain`.  

`formData` and `payload` remain mutually exclusive when submitting an `HTMLElement`, `FormData`, or `URLSearchParams` instance.  But not when you submit a raw object.